homeassistant/components/monoprice/media_player.py

```diff
@@ -156,6 +156,11 @@ def update(self):
             self._source = None
         return True
 
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self._zone_id
```

Comment:
This doesn't work because one could configure two devices via 2 different serial ports. Can you get a unique ID from the device ?

Reply:
The device does not have a unique id. I thought about using the serial port address as the prefix, but:
1. The current implementation also doesn't seem to support more than one instance (or at least has confusing functionality with snapshots).
2. These devices can chain together if you have 2 or 3, so having two unique devices, each connected via serial, should be very rare.

Reply:
And of course that would mean that a change to the serial address would create new entities.

Reply:
We need a better solution for Home Assistant to deal with these cases. In it's current form we cannot accept this PR, as it means we might have to break it in the future when it will be possible to get unique IDs per device.

Reply:
OK, I'm open to other ideas. Here's my thinking:

Some axioms first:
* The device itself does not have any unique ID
* The zones within it do have a unique ID relative to the device
* If you have several of them, up to 3 devices can be chained together so they look like a single device to HA. A deployment with two distinct instances seems like an extremely rare occasion.
* The existing implementation wasn't designed for multiple instances, but might partially work (I don't really have a way to test it).

Here are my humble suggestions:
1. Create some sort of mechanism for "singleton" devices that can only have one instance in HA. Some other components can clearly fit this category (e.g. Sun), and I'm pretty sure there are also a lot of integrations that might not support multiple instances (many integrations in the "Hub" category are likely candidates, for example). This will limit functionality, but ensure no errors are caused as a result of multiple instances in integrations that don't support it. This will require core changes though, and can not be implemented just within this PR.
2. Add the serial address as a prefix to the unique ID. As mentioned, this will mean new entities if the address changes (but then again, if this was configured through a config entry, a change in address would likely have caused new entities as well).
3. Allow the user to define a prefix per instance. In essence, this means the user is controlling the unique id of each device. This is the simplest and cleanest implementation, I believe, and could be generalized as a pattern.
4. Once the configuration switches over to a config entry, store a random value in the config entry as a prefix. As long as it's the same config entry, the entities will stay the same.

What do you think?